### PR TITLE
[NeMo-UX] Add missing docstrings and update some defaults

### PR DESCRIPTION
**Pull request by ashors1**
# What does this PR do ?
- Adds/updates docstrings for the following classes:
    - `PreTrainingDataModule`
    - `MegatronParallel`
    - `MegatronStrategy`
    - `ModelCheckpoint`
    - `Resume`
- Updates the default of `ckpt_include_optimizer` to `True`
- Makes `model` an optional argument to `nemo_path` as the model is only required when `adapter_path` is provided

**Collection**: LLM

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/lightning/pytorch/strategies.py [hunk 1]
```diff
@@ -100,19 +123,19 @@ def __init__(
...
-        ckpt_torch_dist_multiproc=None,  ## TODO(ashors): put elsewhere?
-        ckpt_assume_constant_structure=False,
-        ckpt_parallel_save=True,
-        ckpt_parallel_save_within_dp=False,
-        ckpt_parallel_load=False,
-        ckpt_parallel_save_optim=True,
-        ckpt_load_directly_on_device=True,
+        save_ckpt_format: str = 'torch_dist',
+        ckpt_async_save: bool = False,
+        ckpt_torch_dist_multiproc: int = None,  ## TODO(ashors): put elsewhere?
```
Comment (mikolajblaz) [inline nemo/lightning/pytorch/strategies.py:132]: Is there a plan to leave those args here or actually move elsewhere?
How will they be configured?

**Review by hemildesai — DISMISSED** [review]

File: nemo/lightning/pytorch/strategies.py (hunk 1 above)
Comment (ashors1) [reply nemo/lightning/pytorch/strategies.py:132]: I think we'll leave them here for now, because finding a cleaner solution is nontrivial. They can be configured when the user creates their strategy, e.g. 
```
strategy = nl.MegatronStrategy(
    ckpt_save_format='zarr',
    ckpt_parallel_save_optim=False,
    ckpt_async_save=False,
)
```

Comment (mikolajblaz) [reply nemo/lightning/pytorch/strategies.py:132]: is it configurable with a yaml config?

**Review by mikolajblaz — DISMISSED** [review]

**Review by athitten — APPROVED** [review]
LGTM from the datamodule docstrings side. Thank you @ashors1

**Review by hemildesai — APPROVED** [review]
